Dockerfile

```
@@ -20,13 +19,24 @@
     && rm -rf /var/lib/apt/lists/* \
     && curl -sSL https://install.python-poetry.org | python3 - --version ${POETRY_VERSION}
 
+# Create and set up virtual environment directory
 RUN mkdir /venv
 WORKDIR /venv
-COPY pyproject.toml .
+COPY pyproject.toml poetry.lock ./
+
+# Install dependencies
 RUN poetry config virtualenvs.in-project true
+RUN poetry lock
```

Hadolint (2.12.0)

[info] Multiple consecutive `RUN` instructions. Consider consolidation.

(DL3059)